driver-core/src/test/java/com/datastax/driver/core/ClusterDelegateTest.java

```diff
@@ -0,0 +1,135 @@
+package com.datastax.driver.core;
+
+import java.net.InetSocketAddress;
+import java.util.Collection;
+
+import com.google.common.collect.Lists;
+import org.testng.annotations.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ClusterDelegateTest extends CCMBridge.PerClassSingleNodeCluster {
+    @Override
+    protected Collection<String> getTableDefinitions() {
+        return Lists.newArrayList();
+    }
+
+    @Test(groups = "short")
+    public void should_allow_subclass_to_delegate_to_other_instance() {
+        DelegatingCluster delegatingCluster = new DelegatingCluster(cluster);
+
+        ResultSet rs = delegatingCluster.connect().execute("select * from system.local");
+
+        assertThat(rs.all()).hasSize(1);
+    }
+
+    @Test(groups = "short", expectedExceptions = IllegalStateException.class)
+    public void should_fail_if_delegating_subclass_has_not_overriden_public_method() {
+        BrokenDelegatingCluster brokenCluster = new BrokenDelegatingCluster(cluster);
+        brokenCluster.connect();
+    }
+
+    /**
+     * Demonstrates how to write a {@link Cluster} subclass that wraps another instance (delegate / decorator pattern).
+     *
+     * Cluster should really be an interface (which would have made this process trivial), but for historical reasons
```

Comment:
This is pretty awkward, but understand its to cover a certain need.  I think as a follow on task for [JAVA-616](https://datastax-oss.atlassian.net/browse/JAVA-616) we should make this an interface.